diff --git a/scripts/generate_builtins.py b/scripts/generate_builtins.py
index 98b88f1..02da6b8 100755
--- a/scripts/generate_builtins.py
+++ b/scripts/generate_builtins.py
@@ -27,6 +27,10 @@ _VERBATIM_RE = re.compile(r'^\\\$\n(.*?)^\\\$', re.DOTALL | re.MULTILINE)


 def _ensure_gn(out_dir: str):
+    if "NIX_GN_SRC" in os.environ:
+        os.symlink(src=os.environ["NIX_GN_SRC"], dst=os.path.join(out_dir, 'gn'))
+        return
+
     gn_dir = os.path.join(out_dir, 'gn')
     if not os.path.exists(gn_dir):
         subprocess.check_call(['git', 'clone', 'https://gn.googlesource.com/gn'], cwd=out_dir)
